pyworkflow/pyworkflow/workflow.py

```diff
@@ -5,54 +5,43 @@
 from .node import Node
 from .node_factory import node_factory
 
-from django.conf import settings
-
 
 class Workflow:
     """ Workflow object
 
     Attributes:
+        name: Name of the workflow
+        root_dir: Used for reading/writing files to/from disk
         graph: A NetworkX Directed Graph
-        file_path: Location of a workflow file
+        flow_vars: Global flow variables associated with workflow
     """
 
-    def __init__(self, graph=nx.DiGraph(), file_path=None, name='a-name', flow_vars=nx.Graph(), root_dir=settings.MEDIA_ROOT):
-        #TODO: need to discuss a way to generating the workflow name. For now passing a default name.
-        self._graph = graph
-        self._file_path = file_path
+    def __init__(self, name="Untitled", root_dir=None, graph=nx.DiGraph(), flow_vars=nx.Graph()):
         self._name = name
-        self._flow_vars = flow_vars
+
+        if root_dir is None:
+            root_dir = os.getcwd()
 
         if not os.path.exists(root_dir):
             os.makedirs(root_dir)
+
         self._root_dir = root_dir
 
+        self._graph = graph
+        self._flow_vars = flow_vars
+
     @property
     def graph(self):
         return self._graph
 
-    @graph.setter
-    def graph(self, graph):
-        self._graph = graph
-
-    def path(self, file_name):
-        return os.path.join(self.root_dir, file_name)
+    @staticmethod
+    def path(workflow, file_name):
```

Comment:
If we make this method static, should we create a utils class concerned with all the file handling? Maybe we can add also a method for determining the root_dir value for the workflow which can handle the logic in lines 21 to 29?

Reply:
I went with making `path()` and instance method because it seemed cleaner to me. I really like your idea about adding a utils class, and move the `root_dir` logic there.

`retrieve_node_data` and `store_node_data` might work better as Util methods too, but for now I'm leaving them as static within `Workflow` as we work more on passing data between nodes, but I think this is something we might revisit soon.